ui/src/vayla-design-lib/dialog/dialog.tsx

```diff
@@ -32,14 +32,51 @@ export const Dialog: React.FC<DialogProps> = ({
         scrollable && styles['dialog--scrollable'],
     );
 
+    const dialogHeaderRef = React.useRef<HTMLDivElement>(null);
+    const [inMovingMode, setInMovingMode] = React.useState(false);
+    const [moved, setMoved] = React.useState(false);
+    const [x, setX] = React.useState(0);
+    const [y, setY] = React.useState(0);
+
+    const moveDialog = (e: React.MouseEvent) => {
+        if (inMovingMode) {
+            setX(x + e.movementX);
+            setY(y + e.movementY);
+        }
+    };
```

Comment:
Täällä varsinaista logiikkaa. Dialogilla on ns. "moving mode", johon mennään `onMouseDown`:illa headerissa ja josta tullaan pois kun hiiri nousee ylös. `onMouseUp` ja `onMouseMove` on rekisteröity koko overlaylle, sillä ennen siirtoa hiiri voi siirtyä header-osuuden ulkopuolelle jos liikuttaa hiirtä tarpeeksi nopeasti, ja tällöin ne eivät laukea headerille.

Reply:
Jostain syystä ikkuna tuntuu aika herkästi lagaavan vähän kursorin perässä ja joskus päätyvän hiukan epäsynkkaan. Ei varmaankaan kovin merkityksellistä, mutta kiinnitti huomion.

Reply:
Tein pikakokeiluna pyörityksen jossa ikkuna viedään aina kursorin kohdalla eikä käytetä deltoja. Ainakin minun koneella tuntuu toimivan hiukan paremmin. Silti siinä on pieni viive, mutta ikkuna ei joudu koskaan epäsynkkaan vaikka raahailisi ihan ruudun reunalle asti: https://github.com/finnishtransportagency/geoviite/pull/214